components/expenses/PayoutBankInformationForm.js

```diff
@@ -229,6 +229,12 @@ const DetailsForm = ({ disabled, getFieldName, formik, host, currency }) => {
         {error.message && `: ${error.message}`}
       </MessageBox>
     );
+  } else if (!data?.host?.transferwise) {
+    return (
+      <MessageBox fontSize="12px" type="warning">
+        Wise is not configured on the platform.
+      </MessageBox>
+    );
```

Comment:
I don't see the justification for this, can we simply rely on the one below (line 372-376) instead?

Reply:
Out of clarity, maybe we have to change the condition, for sure.

Reply:
You mean changing the condition at line 372 to handle both cases? We can do that but then we are moving back into a generic solution which we did in https://github.com/opencollective/opencollective-frontend/pull/6742 which didn't work. Over there we are checking for the `availableCurrencies` and instead of returning `null` which is why [you saw the empty form](https://github.com/opencollective/opencollective-frontend/pull/6831#issuecomment-913001919) we are now rendering the warning message. Over here we are trying to prevent the crash if `transferwise` is not defined. I am thinking the justification for this here is to prevent the imminent crash if `transfer-wise` is not defined,

https://github.com/opencollective/opencollective-frontend/blob/dd6a2172d9589e6555913de761d29b7557db4ce8/components/expenses/PayoutBankInformationForm.js#L244

whereas over there it is more specifically aimed at the making sure the form is not displayed (when submitting expenses) when there's no `availableCurrencies`. Please correct me if I am wrong on this one. 🤔 

On another note, I just pushed a change to avoid duplicating the warning message by including it in a function instead.

Reply:
> I'm personally lost. I don't get why we can't handle this in a single place. If `PayoutBankInformationForm` is calling `DetailsForm` and `DetailsForm` is not used anywhere else, why do we have to handle this in both `PayoutBankInformationForm` and `DetailsForm`? There is something wrong.
> 
> Please review the conditions and explain the cases you're trying to handle, add comments in your code to document your knowledge of the problem.

@znarf : Sorry, I was confused on this a bit I think yesterday; yes you are right we can handle this in a single place. I've added the checking for Wise condition to the `PayoutBankInformationForm` and now we have all of it at a single place; seems to work fine. Now I've added a small comment as well on top of the if-else statement. Let me know how it goes. 🙂